src/main/kotlin/be/bluexin/drpc4k/jna/RPCActor.kt

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2019 Arnaud 'Bluexin' Solé
+ *
+ * This file is part of drpc4k.
+ *
+ * drpc4k is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * drpc4k is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with drpc4k.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package be.bluexin.drpc4k.jna
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.channels.actor
+import mu.KotlinLogging
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Start a new Discord RPC Actor in the current [this] [CoroutineScope].
+ *
+ * The newly created actor will receive messages of type [RPCInputMessage] and send [RPCOutputMessage] to the
+ * specified [output].
+ *
+ * A typical usage looks like this :
+ * ```
+ * val rpcOutput = Channel<RPCOutputMessage>(capacity = Channel.UNLIMITED)
+ * val rpcInput = rpcActor(rpcOutput)
+ * // Connect to the client via RPC
+ * rpcInput.send(RPCInputMessage.Connect(myClientKey))
+ * // Update rich presence
+ * rpcInput.send(RPCInputMessage.UpdatePresence(myPresence))
+ * // Set up receiving of updates from the RPC actor
+ * launch {
+ *     for (msg in rpcOutput) with(msg) {
+ *         when (this) {
```

Comment:
why `with(msg)` instead of simply `when(msg)`?

Reply:
Because msg is the only thing used in the context, so `this` can get smart casted in the message handlers, which just makes it a bit shorter.